lib/youtube-dl.js

```diff
@@ -246,23 +237,17 @@ ytdl.getFormats = function(url, args, callback) {
     callback = args;
     args = [];
   }
-  call(url, ['--list-formats'], args, null, function(err, data) {
+  ytdl.getInfo(url, args, {}, function (err, video_info) {
     if (err) return callback(err);
 
-    var formats = [];
-    var status = '';
-
-    data.map(function(line) {
-      var result = formatsRegex.exec(line);
-      if (/\[info\]/.test(line)) { status = line.split(' ')[4].slice(0, -1); }
-      if (result) {
-        formats.push({
-          id         : status,
-          itag       : parseInt(result[1], 10),
-          filetype   : result[2],
-          resolution : result[3],
-        });
-      }
+    var formats_info = video_info.formats || [video_info];
```

Comment:
Yes, some extractors don't define it and instead store the `url`, `ext` and other formats fields in the main dictionary. This line of code is what youtube-dl uses when processing the formats (like with the `--list-formats`). If you don't like this line, I guess I can add the field in youtube-dl.